src/components/AccountDeleteModal.tsx

```diff
@@ -1,23 +1,111 @@
-import { Button, ModalProps } from 'react-bootstrap'
+import { Button, ModalProps, Spinner } from 'react-bootstrap'
 import PopUp from './PopUp'
+import { useState } from 'react'
+import { useForm } from 'react-hook-form'
+import Input from './Input'
+import { ApiUsers } from 'src/api'
+import { toast } from 'react-hot-toast'
+import { useAuth } from 'src/contexts/auth/AuthContext'
 
 interface IAccountDeleteModal extends ModalProps {}
 
 export default function AccountDeleteModal({
   ...modalProps
 }: IAccountDeleteModal) {
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false)
+  const { setUser } = useAuth()
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting },
+  } = useForm({ mode: 'onSubmit', defaultValues: { password: '' } })
+
+  const handleShowConfirmPassword = () => {
+    setShowConfirmPassword(true)
+  }
+  const handleCloseConfirmPassword = () => {
+    setShowConfirmPassword(false)
+    modalProps.onHide && modalProps.onHide()
+  }
+
+  const onSubmit = async (data: { password: string }) => {
+    const userPassword = data.password
+
+    await ApiUsers.destroy(userPassword)
+      .then(() => {
+        toast.success('Account successfully deleted')
+        setUser(undefined)
+      })
+      .catch(() => {
+        toast.error('Incorrect password entered. Please try again.')
+      })
+  }
+  const passwordErrors = errors?.password?.message
+
   return (
-    <PopUp closeButton title="Delete account?" {...modalProps}>
-      <p>
-        Do you wish to delete your account and any reviews you have created?
-        This action cannot be undone.
-      </p>
-      <div className="d-flex flex-row justify-content-between">
-        <Button variant="tertiary" onClick={modalProps.onHide}>
-          Cancel
-        </Button>
-        <Button variant="danger">Delete my account</Button>
-      </div>
+    <PopUp
+      closeButton
+      title="Delete account?"
+      {...modalProps}
+      onHide={handleCloseConfirmPassword}
+    >
+      {!showConfirmPassword ? (
```

Comment:
The flickering is happening because the `handleCloseConfirmPassword` is coupling `setShowConfirmPassword` with the `onHide` function. `onHide()` triggers the modal animation which takes a few hundred milliseconds, but it also calls `setShowConfirmPassword(false)` which instantly reverts the modal's internal state, without waiting for the animation to finish.

The simple solution would be to ditch the `handleCloseConfirmPassword`, which allows `onHide` to be passed with `modalProps` per usual. Then use the `onExited` prop on the modal, which is a callback that fires after the animation finishes. e.g.

```
    <PopUp
      closeButton
      title="Delete account?"
      {...modalProps}
      onExited={() => setShowConfirmPassword(false)}
    >
```
I found this in the react-bootstrap docs which describes onExited and other optional callback props: https://react-bootstrap.netlify.app/docs/components/modal#modal. I reference these docs often, as the library supports lots of use cases like this.